### PR TITLE
Better error message if choice not found

### PR DESCRIPTION
**Pull request by benrudolph**
This throws a more descriptive error message if the choice is not found
in the list of choices for a ChoiceList

As discussed @czue 
screen shot 2015-02-23 at 11 29 36 am

http://manage.dimagi.com/default.asp?157835#877927

### CODE REVIEW

**Comment by dimagimon** [comment]
:+1: No lint errors found.

File: corehq/apps/reports_core/filters.py [hunk 1]
```diff
@@ -198,6 +198,11 @@ def __init__(self, name, required=True, label='Choice List Filter',
 
     def value(self, **kwargs):
         choice = unicode(kwargs[self.name])
+        choice_values = map(lambda c: c.value, self.choices)
+        if choice not in choice_values:
+            raise FilterValueException(u'Choice "{choice}" not found in choices: {choices}'
```
Comment (czue) [inline corehq/apps/reports_core/filters.py]: should be marked for translation

**Comment by czue** [comment]
looks good pending one comment

**Comment by dimagimon** [comment]
:+1: No lint errors found.

**Pull request merged**